src/db/index.js

```diff
@@ -283,5 +285,40 @@ module.exports = {
         console.error(e)
         throw 'Failed to update or add an alert'
       })
+  },
+  saveBlockedReportUser: async function(userId, username) {
+    try {
+      const filter = { userId };
+      const result = await BlockedReport.findOneAndUpdate(filter, { $setOnInsert: {
+          userId,
+          username
```

Comment:
> Hey, I've been on kayaking trip so haven't had time to do a review up until now. Thank you for this thoughtful PR, I've ran through this code once and added some changes(mostly copywriting suggestions)
> 
> Also tried running this on a server without community rules set up and it didn't seem to fail gracefuly, just popped off some messaging and errors in the logs.
> ![image](https://user-images.githubusercontent.com/11031051/131254642-c82becd9-b722-4a89-b568-38fce043a9cc.png)
> 
> Going to do more testing now.

I didn't implement that, currently it uses "fetchRules()" but we need something like an interface that fallback to custom rules if it's not a community server: like we need a new command to set up custom rules in the bot, or maybe make the bot use a #rules channel and parse rules from there, need to think more about it

Reply:
> but we need something like an interface that fallback to custom rules if it's not a community server:

I think for MVP it's better to just leave servers without rules unsupported, as /rules command doesn't support non community servers either so that would also be consistent with it.